### ISSUE TITLE
UIStackView breaks constants ?

### ISSUE THREAD

**Issue by AlexMaTo** [schibsted/layout#102]
Hello,

I'm trying to create a really basic UI in a xml file, something like this :

	<UIStackView
		width="min(auto + 20, parent.width - 20)"
		height="auto"
		left="50% - width / 2"
		top="50% - height / 2"
		alignment="center"
		axis="vertical"
		spacing="8">
			
		<UILabel
			width="auto"
			height="auto"
			top="50% - height / 2"
			left="previous.right"
			textAlignment="center"
			text="{attachmentName}"
			backgroundColor="#0F0" />
	</UIStackView>

In my controller, I load the layout file like that :

	loadLayout(
	    named: "MyLayout.xml",
	    constants: [
	        "attachmentName": attachment.name
	    ]
	)

When I run it, I got a red error message "Unknown property 'attachmentName' of UILabel in 'text' expression......."
Before the red message pops, I can see the view which looks fine... 

Note that if I remove the UIStackView, it's working

Pretty weird, am I doing something wrong ?

**Comment by nicklockwood** [comment]
@AlexMaTo thanks for reporting. It looks like a bug in Layout - I'll take a look.

**Comment by AlexMaTo** [comment]
Hum... Unfortunately I'm facing the same issue using the version 0.6.18, is it the last one ?

I got around the issue by using simple UIView, not very clean but, eh, it's working.

I put the XML file as an attachment, it's not really clean as I just replaced the UIView with UIStackView but it's just for test purpose.

[AttachmentDetailViewV2.xml.txt](https://github.com/schibsted/layout/files/1846573/AttachmentDetailViewV2.xml.txt)

**Comment by wpcfan** [comment]
I just change the example xml, and run into this

```xml
<BoxesViewController
    outlet="layoutNode"
    view.backgroundColor="#151d2b">

    <UIImageView
        height="auto"
        image="Home_Head_Background"
        top="0"
        width="100%"
    />
    <UIScrollView
        contentInset.bottom="safeAreaInsets.bottom + 15"
        contentInset.left="safeAreaInsets.left + 15"
        contentInset.right="safeAreaInsets.right + 15"
        contentInset.top="previous.center.y + 30"
        contentInsetAdjustmentBehavior="never"
        scrollIndicatorInsets.bottom="safeAreaInsets.bottom + 15"
        scrollIndicatorInsets.top="previous.center.y + 15">

        <UIStackView
            alignment="center"
            axis="horizontal"
            spacing="15"
            width="parent.width">

            <UIImageView
                image="Home_Bar_Green"
                left="parent.safeAreaInsets.left"
                top="parent.safeAreaInsets.top"
                width="(parent.width - 60) / 5">

                <UILabel
                    font="12"
                    left="parent.left + 8"
                    text="{strings.home.bar.temperature}"
                    top="parent.top + 6"
                />
                <UILabel
                    left="previous.left + 6"
                    textColor="#fff"
                    top="previous.bottom + 6">

                    <b>23</b>{temperatureString}
                </UILabel>
            </UIImageView>
            <UIImageView
                image="Home_Bar_Green"
                left="previous.right + 15"
                top="previous.top"
                width="(parent.width - 60) / 5">

                <UILabel
                    font="12"
                    left="parent.left + 8"
                    text="{strings.home.bar.humity}"
                    top="parent.top + 6"
                />
                <UILabel
                    left="previous.left + 6"
                    textColor="#fff"
                    top="previous.bottom + 6">

                    <b>23</b>{percentageString}
                </UILabel>
            </UIImageView>
            <UIImageView
                image="Home_Bar_Green"
                left="previous.right + 15"
                top="previous.top"
                width="(parent.width - 60) / 5">

                <UILabel
                    font="12"
                    left="parent.left + 8"
                    text="{strings.home.bar.pm25}"
                    top="parent.top + 6"
                />
                <UILabel
                    font="20 bold"
                    left="previous.left + 6"
                    text="23"
                    textColor="#fff"
                    top="previous.bottom + 6"
                />
            </UIImageView>
            <UIImageView
                image="Home_Bar_Green"
                left="previous.right + 15"
                top="previous.top"
                width="(parent.width - 60) / 5">

                <UILabel
                    font="12"
                    left="parent.left + 10"
                    text="{strings.home.bar.cq2}"
                    top="parent.top + 6"
                />
                <UILabel
                    font="20 bold"
                    left="previous.left + 6"
                    text="23"
                    textColor="#fff"
                    top="previous.bottom + 6"
                />
            </UIImageView>
            <UIImageView
                image="Home_Bar_Purple"
                left="previous.right + 15"
                top="previous.top"
                width="(parent.width - 60) / 5">

                <UILabel
                    font="12"
                    left="parent.left + 8"
                    text="{strings.home.bar.tvoc}"
                    top="parent.top + 6"
                />
                <UILabel
                    font="20 bold"
                    left="previous.left + 6"
                    text="23"
                    textColor="#fff"
                    top="previous.bottom + 6"
                />
            </UIImageView>
        </UIStackView>
        <UIStackView
            alignment="center"
            axis="horizontal"
            left="previous.left"
            spacing="15"
            top="previous.bottom + 15"
            width="parent.width">

            <!-- First column -->
            <UIStackView
                alignment="center"
                axis="vertical"
                left="parent.left"
                spacing="15"
                width="(parent.width - 15) / 2">

                <UIImageView
                    backgroundColor="#31425c"
                    height="width + 15"
                    id="first"
                    image="Home_Tile_Visitor"
                    layer.cornerRadius="10"
                    left="parent.safeAreaInsets.left"
                    top="previous.bottom + 15"
                    transform.rotation="isToggled ? pi / 2 : 0"
                    width="parent.width">

                    <UILabel
                        font="20"
                        left="parent.left + 10"
                        text="{strings.home.visitor}"
                        textColor="#fff"
                        top="parent.top + 10"
                    />
                    <UILabel
                        font="10"
                        left="previous.left"
                        text="{strings.home.visitor.caption}"
                        textColor="#7c8ca6"
                        top="previous.bottom + 2"
                    />
                </UIImageView>
                <UIImageView
                    backgroundColor="#31425c"
                    height="width / 2"
                    image="Home_Tile_Device"
                    layer.cornerRadius="10"
                    left="previous.left"
                    top="15 + previous.bottom"
                    width="previous.width">

                    <UILabel
                        left="parent.left + 10"
                        text="{strings.home.mydevice}"
                        textColor="#fff"
                        top="parent.top + 10"
                    />
                    <UILabel
                        font="10"
                        left="previous.left"
                        text="{strings.home.mydevice.caption}"
                        textColor="#7c8ca6"
                        top="previous.bottom + 2"
                    />
                </UIImageView>
            </UIStackView>
            

            <!-- Second column -->
            <UIStackView
                alignment="center"
                axis="vertical"
                spacing="15">

                <UIImageView
                    backgroundColor="#31425c"
                    height="width / 2"
                    image="Home_Tile_Meeting"
                    layer.cornerRadius="10"
                    left="width + 15"
                    top="#first.top"
                    width="#first.width">

                    <UILabel
                        left="parent.left + 10"
                        text="{strings.home.meeting}"
                        textColor="#fff"
                        top="parent.top + 10"
                    />
                    <UILabel
                        font="10"
                        left="previous.left"
                        text="{strings.home.meeting.caption}"
                        textColor="#7c8ca6"
                        top="previous.bottom + 2"
                    />
                </UIImageView>
                <UIImageView
                    backgroundColor="#31425c"
                    height="width / 2"
                    image="Home_Tile_Email"
                    layer.cornerRadius="10"
                    left="previous.left"
                    top="15 + previous.bottom"
                    width="previous.width">

                    <UILabel
                        left="parent.left + 10"
                        text="{strings.home.email}"
                        textColor="#fff"
                        top="parent.top + 10"
                    />
                    <UILabel
                        font="10"
                        left="previous.left"
                        text="{strings.home.email.caption}"
                        textColor="#7c8ca6"
                        top="previous.bottom + 2"
                    />
                </UIImageView>
                <UIImageView
                    backgroundColor="#31425c"
                    height="width / 2"
                    image="Home_Tile_More"
                    layer.cornerRadius="10"
                    left="previous.left"
                    top="15 + previous.bottom"
                    width="previous.width">

                    <UILabel
                        left="parent.left + 10"
                        text="{strings.home.more}"
                        textColor="#fff"
                        top="parent.top + 10"
                    />
                    <UILabel
                        font="10"
                        left="previous.left"
                        text="{strings.home.more.caption}"
                        textColor="#7c8ca6"
                        top="previous.bottom + 2"
                    />
                </UIImageView>
            </UIStackView>
        </UIStackView>
    </UIScrollView>

    <!-- Floating button -->
    <UIView
        backgroundColor="#fff9"
        bottom="previous.bottom + parent.safeAreaInsets.bottom"
        height="50"
        left="0"
        width="100%">

        <UIButton
            backgroundColor="isToggled ? colors.blue : colors.red"
            bottom="10"
            layer.cornerRadius="5"
            left="safeAreaInsets.left + 15"
            right="safeAreaInsets.right + 15"
            title="Toggle"
            top="10"
            touchUpInside="{isToggled ? 'setUntoggled' : 'setToggled'}"
        />
    </UIView>
</BoxesViewController>

```

strings
 
```ini
"Lorem.Ipsum" = "Lorem ipsum dolor sit amet, consectetur adipiscing elit.";
"home.mydevice" = "我的设备";
"home.mydevice.caption" = "SMART DEVICES";
"home.visitor" = "访客邀请";
"home.visitor.caption" = "VISITOR INVITATIONS";
"home.meeting" = "会议预定";
"home.meeting.caption" = "SCHEDULES";
"home.email" = "电子邮件";
"home.email.caption" = "EMAIL";
"home.more" = "更多应用";
"home.more.caption" = "MORE";
"home.bar.temperature" = "温度";
"home.bar.humity" = "湿度";
"home.bar.pm25" = "PM2.5";
"home.bar.cq2" = "CQ2";
"home.bar.tvoc" = "TVOC";
```

**Comment by nicklockwood** [comment]
@AlexMaTo thanks to @wpcfan's example, I was able to reproduce this. It should be fixed in 0.6.28.
